File: data/readable_format/cpp/file596.cpp
#include <iostream>
using namespace std;

int main() {
    int n, result;

    cout << "Enter a positive integer" << endl;
    cin >> n;

    result = 1;
    while (n > 1) {
        result *= n;
        --n;
    }

    cout << "Factorial is: " << result << endl;

    return 0;
}

File: data/readable_format/fortran/file596.f90
program factorial
    implicit none
    integer :: n, result

    print *, 'Enter a positive integer'
    read *, n

    result = 1
    do while (n > 1)
        result = result * n
        n = n - 1
    end do

    print *, 'Factorial is: ', result
end program factorial